pre_commit/languages/coursier.py

```diff
@@ -27,6 +27,14 @@ def install_environment(
     helpers.assert_version_default('coursier', version)
     helpers.assert_no_additional_deps('coursier', additional_dependencies)
 
+    # Support both possible executable names (either "cs" or "coursier")
+    executable = next(filter(helpers.exe_exists, ('cs', 'coursier')), None)
```

Comment:
unnecessary FP -- we avoid `map` / `filter` / `reduce` 

this should be `exe('cs') or exe('coursier')` once you find the right function (`helpers.exe_exists` isn't it -- that returns a `bool`)

Reply:
> this should be `exe('cs') or exe('coursier')` once you find the right function (`helpers.exe_exists` isn't it -- that returns a `bool`)

Do you want me to write a new one or is there something that I can use? `shutil.which` would be a candiate, or maybe `pre_commit.parse_shebang.find_executable` which kind of duplicates that functionality.

Reply:
`find_executable` is the right one probably yes